esp_delta_ota/examples/https_delta_ota/tools/esp_delta_ota_patch_gen.py

```diff
@@ -19,40 +23,120 @@
 # Magic Byte is created using command: echo -n "esp_delta_ota" | sha256sum
 esp_delta_ota_magic = 0xfccdde10
 
-MAGIC_SIZE = 4
-DIGEST_SIZE = 32
-RESERVED_HEADER = 64 - (MAGIC_SIZE + DIGEST_SIZE)
+MAGIC_SIZE = 4 # This is size of magic byte
+DIGEST_SIZE = 32 # This is SHA256 of the base binary    
+HEADER_VERSION_SIZE = 1 # This is size of header version
+CRC_SIZE = 4 # This is CRC32 of patch file which shall be applied on the base binary
+HEADER_SIZE = 64
+RESERVED_HEADER = HEADER_SIZE - (MAGIC_SIZE + DIGEST_SIZE + HEADER_VERSION_SIZE + CRC_SIZE) # This is reserved header size
+
+HEADER_VERSION = 0 # Define the HEADER_VERSION as 0
```

Comment:
Instead of naming this as header version, lets rename this as PATCH_VERSION.

lets define PATCH_VERSION = 1 in this case.

That mean all of the previous patches are considered as 0 and now a patch which has this version field shall be considered as version 1